common/reviews/api/rush-lib.api.md

```diff
@@ -1384,7 +1384,7 @@ export class RushLifecycleHooks {
 // @alpha
 export class RushProjectConfiguration {
     readonly disableBuildCacheForProject: boolean;
-    getCacheDisabledReason(trackedFileNames: Iterable<string>, phaseName: string): string | undefined;
+    getCacheDisabledReason(operation: Operation, trackedFileNames: Iterable<string>, phaseName: string): string | undefined;
```

Comment:
This is technically a breaking change, even though the API is alpha

Reply:
If I flipped it to 
```
getCacheDisabledReason(trackedFileNames: Iterable<string>, phaseName: string, operation?: Operation): string | undefined;
```
that would be less breaking right?

Reply:
@iclanton I'm not sure I agree with this recommendation.  It seems like in the current the `getCacheDisabledReason()` is not going to produce correct results unless the `Operation` is included, so it is arguably a better experience for people's plugins to have a compile error about a missing function parameter, rather than silently appearing to compile while producing a wrong result.  (In general, we should be cautious about introducing optional parameters in public APIs, because they frequently produce error-prone contracts.)

Is there actually a realistic concern about third-party Rush plugins using this `@alpha` API?  If so, as a compromise what we could do is make `operation` into a required parameter, but put it last and retain the `operation?.runner` test in case an old plugin calls the API incorrectly (because it was compiled using an older Rush .d.ts file).

(Although it's questionable how much risk there is for this contract, given that it is  `@alpha`.)

Reply:
Following David's suggestion, this now takes in a `isNoOp: boolean`, which is a breaking change but as Pete points out, is needed to ensure we product correct results.